libsolidity/formal/SMTEncoder.cpp

```diff
@@ -869,6 +869,9 @@ void SMTEncoder::endVisit(IndexAccess const& _indexAccess)
 {
 	createExpr(_indexAccess);
 
+	if (_indexAccess.annotation().type->category() == Type::Category::TypeType)
+		return;
+
 	shared_ptr<smt::SymbolicVariable> array;
 	if (auto const& id = dynamic_cast<Identifier const*>(&_indexAccess.baseExpression()))
```

Comment:
While you're at it, you could change that to a ``auto const*`` - the ``&`` is really misleading...

Reply:
Ok!